src/bosh_azure_cpi/lib/cloud/azure/storage/storage_account_manager.rb

```diff
@@ -135,14 +135,23 @@ def default_storage_account
 
       storage_account_name = @azure_config.storage_account_name
       unless storage_account_name.nil?
-        @logger.debug("The default storage account is '#{storage_account_name}'")
+        @logger.debug("The default storage account in global settings is '#{storage_account_name}'")
         @default_storage_account = @azure_client.get_storage_account_by_name(storage_account_name)
         cloud_error("The default storage account '#{storage_account_name}' is specified in Global Configuration, but it does not exist.") if @default_storage_account.nil?
         @azure_client.update_tags_of_storage_account(storage_account_name, STEMCELL_STORAGE_ACCOUNT_TAGS) if @use_managed_disks && !is_stemcell_storage_account?(@default_storage_account[:tags])
         return @default_storage_account
       end
 
       @logger.debug('The default storage account is not specified in global settings.')
+      storage_account_name = get_storage_account_name_from_cache
+      unless storage_account_name.empty?
+        @logger.debug("The default storage account from cache is '#{storage_account_name}'")
+        @default_storage_account = @azure_client.get_storage_account_by_name(storage_account_name)
+        cloud_error("The default storage account '#{storage_account_name}' does not exist.") if @default_storage_account.nil?
+        @azure_client.update_tags_of_storage_account(storage_account_name, STEMCELL_STORAGE_ACCOUNT_TAGS) if @use_managed_disks && !is_stemcell_storage_account?(@default_storage_account[:tags])
```

Comment:
i guess it is better to update tag before Line164, so we don't need to tag it every time